test/externalTests/common.sh

```diff
@@ -117,6 +117,19 @@ function neutralize_package_json_hooks
     sed -i 's|"prepare": *".*"|"prepare": ""|g' package.json
 }
 
+function neutralize_packaged_contracts
```

Comment:
What does `neutralize` mean in this context? I'm wondering if the name could be more readable?

Reply:
"Make them not affect the compilation of contracts in the external project". In this particular case it's that some contracts installed with npm come with a framework config file and/or already compiled JSON artifacts with bytecode.This may or may not affect the contracts in the project but I'm just not taking chances that Hardhat would reuse their already compiled bytecode. We have a check that all build artifacts in the directory tree must come from our selected compiler binary and this trips up that check.

I could call it something like `remove_nested_framework_config_and_artifacts` but the current name conveys my intention a bit better IMO. I.e. if there's anything else needed to make these contracts not interfere (like removing their `package.lock` or something), it's something I'd add to this function and then the name would no longer be precise.

Reply:
Sounds good. I'd still reccommend something like `clean_packaged_contracts` that references Makefile's clean command. However, I'd be fine if you would want to go with the current name.

Reply:
I'd rather stay with the current one because I think it's better that it sounds more like workaround than something you'd do normally. If the name makes you uneasy, it works as intended :)